Source/WebCore/Modules/applepay/PaymentCoordinator.cpp

```diff
@@ -107,6 +107,11 @@ bool PaymentCoordinator::beginPaymentSession(Document& document, PaymentSession&
     if (!showPaymentUI)
         return false;
 
+#if ENABLE(APPLE_PAY_SHIPPING_CONTACT_EDITING_MODE)
+    if (auto& shippingContactEditingMode = paymentRequest.shippingContactEditingMode(); shippingContactEditingMode && *shippingContactEditingMode == ApplePayShippingContactEditingMode::Enabled)
```

Comment:
Since this is a tiny object (under 8 bytes), it's probably more efficient to just use `auto` here. You can also just check against `Enabled` without checking for `nullopt` first.
```suggestion
    if (paymentRequest.shippingContactEditingMode() == ApplePayShippingContactEditingMode::Enabled)
```

Reply:
TIL that `bool operator==( const optional<T>& opt, const T& value)` does the `std::nullopt` check first!

Reply:
Thanks for the suggestion, will fix

Reply:
Fixed in [15c78ff](https://github.com/WebKit/WebKit/commit/15c78ff5c1270e6244769c3d64063a94274958a6)